includes/class.llms.frontend.forms.php

```diff
@@ -132,37 +191,123 @@ function mark_course_complete( $user_id, $lesson_id ) {
 			$update_user_postmeta = $wpdb->insert($wpdb->prefix . 'lifterlms_user_postmeta',
 				array(
 					'user_id' => $user_id,
-					'post_id' => $course->id,
+					'post_id' => $section->id,
 					'meta_key' => $key,
 					'meta_value' => $value,
 					'updated_date' => current_time( 'mysql' ),
 				)
 			);
 
-			do_action( 'llms_course_completed', $user_id, $course->id );
-
+			do_action( 'lifterlms_section_completed', $user_id, $section->id );
 		}
 	}
 
 	/**
-	 * mark section complete
-	 * Called by mark_lesson_complte
+	 * mark track complete
+	 * Called by mark_lesson_complete
 	 *
-	 * If all lessons in section complete mark section as complete.
+	 * If all parts in track are complete mark track as complete.
 	 *
 	 * @param  int $user_id [ID of the current user]
 	 * @param  int $lesson_id [ID of the current lesson]
 	 *
 	 * @return void
+	 * @version 3.2.4
 	 */
-	public function mark_section_complete( $user_id, $lesson_id ) {
-
-		global $wpdb;
+	public function mark_track_complete( $user_id, $lesson_id ) {
 
 		$lesson = new LLMS_Lesson( $lesson_id );
 		$course_id = $lesson->get_parent_course();
+
 		$course = new LLMS_Course( $course_id );
-		$course_syllabus = $course->get_syllabus();
+		/**
+		 * This variable is what will store the list of classes
+		 * for each track that this class is a member of
+		 * @var array
+		 */
+		$courses_in_track = array();
+
+		// Get Track Information
+		// This gets the information about all the tracks that
+		// this course is a part of
+		$tracks = wp_get_post_terms( $course->id,'course_track', array( 'fields' => 'all' ) );
```

Comment:
this can be replaced by a combination of LLMS_Track and LLMS_Student::is_complete() -- possibly just the latter which uses the LLMS_Track class